app/tests/components_tests/test_management_commands.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+from django.core.management import call_command
+from django_capture_on_commit_callbacks import capture_on_commit_callbacks
+
+from grandchallenge.components.models import InterfaceKind
+from tests.cases_tests.factories import ImageFactoryWithImageFile16Bit
+from tests.components_tests.factories import (
+    ComponentInterfaceFactory,
+    ComponentInterfaceValueFactory,
+)
+
+
+@pytest.mark.django_db
+def test_add_overlay_segments(settings):
+    settings.task_eager_propagates = (True,)
+    settings.task_always_eager = (True,)
+
+    im = ImageFactoryWithImageFile16Bit()
+    ci = ComponentInterfaceFactory(
+        title="foo", kind=InterfaceKind.InterfaceKindChoices.SEGMENTATION
+    )
+    ComponentInterfaceValueFactory(interface=ci, image=im)
+
+    im.segments = [2]
+    im.save()
+
+    # This raises a ValidationError, but unfortunetaly that gest swallowed in
+    # the context manager.
```

Comment:
I don't think this is correct. The callback is `apply_async`, which [when invoked](https://docs.celeryq.dev/en/stable/reference/celery.app.task.html#celery.app.task.Task.apply_async) returns an [AsyncResult](https://docs.celeryq.dev/en/stable/reference/celery.result.html#celery.result.AsyncResult) instance, even when using eager task execution. This has a `status` and `traceback` attributes that can be used to determine whether the `ValidationError` was raised e.g.

```python
with capture_om_commit_callbacks() as callbacks:
    ...

for callback in callbacks:
    maybe_task = callback()
    if isinstance(maybe_task, AsyncResult):
        # Do something with maybe_task.traceback
```

Reply:
Nice, I'll look into that. Thank you!